scala3doc/test/dotty/dokka/ScaladocTest.scala

```diff
@@ -0,0 +1,90 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.model.{DModule, WithChildren}
+import org.jetbrains.dokka.pages.RootPageNode
+import org.jetbrains.dokka.testApi.testRunner.{DokkaTestGenerator, TestMethods}
+import org.jetbrains.dokka.testApi.logger.TestLogger
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.jetbrains.dokka.DokkaConfiguration
+import scala.jdk.CollectionConverters.{ListHasAsScala, SeqHasAsJava}
+import org.junit.{Test, Rule}
+import org.junit.rules.{TemporaryFolder, ErrorCollector}
+import java.io.File
+
+abstract class ScaladocTest(val name: String):
+  def assertions: Seq[Assertion]
+
+  private def getTempDir() : TemporaryFolder =
+    val folder = new TemporaryFolder()
+    folder.create()
+    folder
+
+  private def args = Args(
+      name = "test",
+      tastyRoots = Nil ,
+      classpath =  System.getProperty("java.class.path"),
+      None,
+      output = getTempDir().getRoot,
+      projectVersion = "1.0",
+      projectTitle = None,
+      projectLogo = None,
+      defaultSyntax = None,
+      sourceLinks = Nil
+    )
+
+  private def tastyFiles =
+    def collectFiles(dir: File): List[String] = dir.listFiles.toList.flatMap {
```

Comment:
`listFiles` can return null if directory does not exisits so we should check that and print proper error message